src/Sylius/Bundle/CoreBundle/Security/UserPasswordResetter.php

```diff
@@ -27,12 +27,17 @@ public function __construct(
     ) {
     }
 
+    /**
+     * @throws UserNotFoundException
+     */
     public function reset(string $token, string $password): void
     {
         /** @var UserInterface|null $user */
         $user = $this->userRepository->findOneBy(['passwordResetToken' => $token]);
 
-        Assert::notNull($user, 'No user found with reset token: ' . $token);
+        if (null === $user) {
+            throw new UserNotFoundException(message: 'Token not found');
```

Comment:
I'm not convinced about the message, imo the one before was more fitting as the user for the specified token was not found, not exactly the token itself 🤔